lib/tasks/npm-task.js

```diff
@@ -124,37 +158,50 @@ class NpmTask extends Task {
    * @method findPackageManager
    * @return {Promise}
    */
-  async findPackageManager() {
-    if (this.useYarn === true) {
+  async findPackageManager({ useYarn } = {}) {
+    if (useYarn === true) {
       logger.info('yarn requested -> trying yarn');
       return this.checkYarn();
     }
 
-    if (this.useYarn === false) {
+    if (useYarn === false) {
       logger.info('npm requested -> using npm');
       return this.checkNpmVersion();
     }
 
-    if (!this.hasYarnLock()) {
-      logger.info('yarn.lock not found -> using npm');
-      return this.checkNpmVersion();
+    if (this.hasYarnLock()) {
```

Comment:
I think we should swap this around to look for `package-lock.json` actually, since `npm@7` supports maintaining a `yarn.lock` as well.

Reply:
if a `yarn.lock` file is detected I would assume the devs expect to use `yarn` to install it. if `yarn` is not available we fall back to `npm`, which in the case of `npm@7` would then also work. the `versionConstraints` in the class say that only npm 3-6 are supported yet though.